### PR TITLE
replaced URLConnection with HttpClient

### PR DESCRIPTION
**Pull request by GrietVermeesch**

### CODE REVIEW

File: autobump-maven/src/main/java/com/github/autobump/maven/model/MavenVersionRepository.java [hunk 1]
```diff
@@ -9,45 +9,64 @@
...
+    }
+
+    private HttpResponse<InputStream> getInputStreamHttpResponse(Dependency dependency) {
+        try {
+            HttpRequest request = HttpRequest.newBuilder().GET().uri(getRepoUri(dependency).toURI())
+                    .build();
+            return httpClient.send(request, HttpResponse.BodyHandlers.ofInputStream());
+        } catch (URISyntaxException e) {
+            throw new WrongUrlException("Wrong URI Syntax", e);
+        } catch (InterruptedException | IOException e) {
```
Comment (timpeeters) [inline autobump-maven/src/main/java/com/github/autobump/maven/model/MavenVersionRepository.java:68]: InterruptedException is always an annoying one. I also see there is a Sonar issue reported about this.

An example of where an InterruptedException might be thrown is for example:
1. The MavenVersionRepository sends a request to Maven Central to download the maven-metadata.xml file and waits for a response.
2. While waiting for the response, Autobump receives a SIGTERM signal to shutdown.
3. Java sends an interrupt to all the active threads including the thread that was waiting for the response.
4. Java interrupts threads by throwing an InterruptedException.
5. What you do in this case largely depends on the use case. 

In this case, I would not catch the exception in the getInputStreamHttpResponse method but at it to the method signature (throws InterruptedException).
I would catch the exception in the getAllAvailableVersions method, and in that catch I would just return an empty set (as if we found no versions at all), and re-interrupt the thread.

        } catch (InterruptedException e) {
            Thread.currentThread().interrupt();

            return Collections.emptySet();
        }

File: autobump-maven/src/main/java/com/github/autobump/maven/model/MavenVersionRepository.java [hunk 2]
```diff
@@ -9,45 +9,64 @@
...
+            }
+        } catch (XmlPullParserException | IOException e) {
+            throw new DependencyParserException("something went wrong while parsing the xml", e);
+        }
+    }
+
+    private HttpResponse<InputStream> getInputStreamHttpResponse(Dependency dependency) {
+        try {
+            HttpRequest request = HttpRequest.newBuilder().GET().uri(getRepoUri(dependency).toURI())
+                    .build();
```
Comment (timpeeters) [inline autobump-maven/src/main/java/com/github/autobump/maven/model/MavenVersionRepository.java:64]: Whenever sending an http request it is a good practice to define a timeout how long you want to wait for a response. E.g. wait at most 5 seconds for a response:

`.uri(...).timeout(Duration.ofSeconds(5)).build()`

File: autobump-maven/src/main/java/com/github/autobump/maven/model/MavenVersionRepository.java [hunk 3]
```diff
@@ -9,45 +9,64 @@
...
+            return httpClient.send(request, HttpResponse.BodyHandlers.ofInputStream());
+        } catch (URISyntaxException e) {
+            throw new WrongUrlException("Wrong URI Syntax", e);
+        } catch (InterruptedException | IOException e) {
+            throw new RuntimeException(e);
```
Comment (timpeeters) [inline autobump-maven/src/main/java/com/github/autobump/maven/model/MavenVersionRepository.java:69]: You might rethrow the IOException as a more specific UncheckedIOException here.

File: autobump-maven/src/main/java/com/github/autobump/maven/model/MavenVersionRepository.java (hunk 1 above)
Comment (GrietVermeesch) [reply autobump-maven/src/main/java/com/github/autobump/maven/model/MavenVersionRepository.java:68]: Yes, I was about to implement the interruption of the thread. Yet, we're considering another implementation now, with an async HttpRequest.

**Pull request closed without merging**